gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -48,13 +62,42 @@ def __init__(self, robust_measurement_noise: bool = True) -> None:
         self._max_1dsfm_projection_directions = MAX_PROJECTION_DIRECTIONS
         self._outlier_weight_threshold = OUTLIER_WEIGHT_THRESHOLD
 
+    def __sample_projection_directions(
+        self, w_i2Ui1_measurements: BinaryMeasurementsUnit3, projection_sampling_method: ProjectionSamplingMethod
+    ) -> List[Unit3]:
+        """Samples projection directions for 1DSfM based on the provided sampling method.
+
+        Args:
+            w_i2Ui1_measurements: Unit translation measurements which are input to 1DSfM.
+            projection_sampling_method: ProjectionSamplingMethod to be used for sampling directions.
+
+        Returns:
+            List of sampled Unit3 projection directions.
+        """
+        num_measurements = len(w_i2Ui1_measurements)
+
+        if projection_sampling_method == self.ProjectionSamplingMethod.SAMPLE_INPUT_MEASUREMENTS:
+            num_samples = min(num_measurements, self._max_1dsfm_projection_directions)
+            sampled_indices = np.random.choice(w_i2Ui1_measurements, num_samples, replace=False)
+            return [w_i2Ui1_measurements[idx].measured() for idx in sampled_indices]
+
+        elif projection_sampling_method == self.ProjectionSamplingMethod.SAMPLE_WITH_INPUT_DENSITY:
+            return _sample_kde_directions(w_i2Ui1_measurements, num_samples=self._max_1dsfm_projection_directions)
+
+        elif projection_sampling_method == self.ProjectionSamplingMethod.SAMPLE_WITH_UNIFORM_DENSITY:
+            return _sample_random_directions(num_samples=self._max_1dsfm_projection_directions)
+
+        else:
+            raise ValueError("Unsupported sampling method!")
```

Comment:
Is this if-else statement unnecessary since there are return statements?

Reply:
not really, doesnt hurt either, I can change later. I thought it would be better to place the exception inside an `else` instead of just default.

Reply:
Maybe define a variable and set it in the if-else statement and return at the end?

Reply:
im ok with it as it is, kind of nice to catch the undefined types

Reply:
I think both are good ways. I am not too sure which one to choose. 
I prefer returning immediately since there is no more relevant code for that branch. The elif ... else.. makes it explicit and slightly easier to understand at first glance. So I'll keep it as is.

Reply:
I think reformatting to the following is preferable to avoid pylint warnings:

```python
if projection_sampling_method == self.ProjectionSamplingMethod.SAMPLE_INPUT_MEASUREMENTS:
    num_samples = min(num_measurements, self._max_1dsfm_projection_directions)
    sampled_indices = np.random.choice(w_i2Ui1_measurements, num_samples, replace=False)
    projections = [w_i2Ui1_measurements[idx].measured() for idx in sampled_indices]
elif projection_sampling_method == self.ProjectionSamplingMethod.SAMPLE_WITH_INPUT_DENSITY:
    projections = _sample_kde_directions(w_i2Ui1_measurements, num_samples=self._max_1dsfm_projection_directions)
elif projection_sampling_method == self.ProjectionSamplingMethod.SAMPLE_WITH_UNIFORM_DENSITY:
    projections = _sample_random_directions(num_samples=self._max_1dsfm_projection_directions)
else:
    raise ValueError("Unsupported sampling method!")

return projections
```

Reply:
done.